pyomo/contrib/pynumero/src/CMakeLists.txt

```diff
@@ -179,6 +191,8 @@ IF( BUILD_MA27 )
   ADD_LIBRARY( pynumero_MA27 SHARED ${PYNUMERO_MA27_SOURCES} )
   IF( MA27_OBJECT )
     TARGET_LINK_LIBRARIES( pynumero_MA27 ${MA27_OBJECT} )
+  ELSEIF( HSL_LIBRARY AND NOT USE_MA27_LIBRARY )
+    TARGET_LINK_LIBRARIES( pynumero_MA27 ${HSL_LIBRARY} )
   ELSE()
     TARGET_LINK_LIBRARIES( pynumero_MA27 ${MA27_LIBRARY} )
   ENDIF()
```

Comment:
I think I would recommend getting rid of the `USE_MA??_LIBRARY` option completely and just:
```cmake
  IF( MA27_OBJECT )
    TARGET_LINK_LIBRARIES( pynumero_MA27 ${MA27_OBJECT} )
  ELSEIF( MA27_LIBRARY )
    TARGET_LINK_LIBRARIES( pynumero_MA27 ${MA27_LIBRARY} )
  ELSE()
    TARGET_LINK_LIBRARIES( pynumero_MA27 ${HSL_LIBRARY} )
  ENDIF()
```

Reply:
I'd prefer to have a way to specify which to use in the case that both libraries are present.

Reply:
Then we should go back to the original implementation: in the original implementation, the user can overwrite MA27_LIBRARY to point to whatever they want: the way cmake's FIND_LIBRARY works, the automated search only happens if the incoming cache variable is _NotFound.

Reply:
Okay, thanks for the reminder, that functionality slipped my mind. Reverted.